modules/ding_wayf_dk/modules/ding_wayf_dk_login/ding_wayf_dk_login.module

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * @file
+ * Adds support for user login through WAYF dk services.
+ */
+
+/**
+ * Delimiter for extracting a part of a field.
+ */
+define('DING_WAYF_DK_LOGIN_FIELD_DELIMITER', ':');
+
+/**
+ * Implements hook_menu_alter().
+ *
+ * Allow users to logout of WAYF as they may be logged into WAYF but not the
+ * drupal site/library system.
+ */
+function ding_wayf_dk_login_menu_alter(&$items) {
+  $items['wayf/logout']['access callback'] = TRUE;
+}
+
+/**
+ * Implements hook_wayf_dk_login_create_user().
+ */
+function ding_wayf_dk_login_wayf_dk_login_create_user($attributes) {
+  if (ding_provider_implements('wayf', 'login_credentials')) {
+    // Get credentials from the provider.
+    $credentials = ding_provider_invoke('wayf', 'login_credentials', ding_wayf_dk_login_get_credentials($attributes));
+
+    // Check if CPR mapping is activated.
+    if (variable_get('ding_wayf_dk_login_debug', 0)) {
+      $credentials['name'] = variable_get('ding_wayf_dk_login_debug_cpr', '');
+    }
```

Comment:
I do not understand what is going on here. The comment refers to "CPR mapping" and uses variables relation to debugging.